### PR TITLE
LPS-136275 use provider instead of repeat code

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
This change is made for this: 

https://github.com/brianchandotcom/liferay-portal/pull/104829#issuecomment-887655932

I tested it with the issue on https://issues.liferay.com/browse/LPS-136206 to check if error is throw or server do not start.

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:license

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

File: modules/apps/image/image-api/src/main/java/com/liferay/image/company/provider/ImageCompanyIdUpgradeProvider.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * @author Alicia Garcia
+ */
+@ProviderType
+public interface ImageCompanyIdUpgradeProvider<T> {
```
Comment (4lejandrito) [inline modules/apps/image/image-api/src/main/java/com/liferay/image/company/provider/ImageCompanyIdUpgradeProvider.java:30]: Hi @AliciaGarciaGarcia,

If you expose the whole upgrade class, you can directly reuse it without wrapping it. That's how [UpgradeDiscussionSubscriptionClassName](https://github.com/liferay/liferay-portal/blob/8072955ec68042ea0ae85320e6c3cbab52cb57fc/modules/apps/comment/comment-page-comments-web/src/main/java/com/liferay/comment/page/comments/web/internal/upgrade/PageCommentsWebUpgrade.java#L60-L66) works. 

What do you think?

**Comment by AliciaGarciaGarcia** [comment]
see https://github.com/liferay-lima/liferay-portal/pull/2135

File: modules/apps/image/image-api/src/main/java/com/liferay/image/company/provider/ImageCompanyIdUpgradeProvider.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/image/image-api/src/main/java/com/liferay/image/company/provider/ImageCompanyIdUpgradeProvider.java:30]: https://github.com/liferay-lima/liferay-portal/pull/2135

**Pull request closed without merging**